src/main/java/com/csoft/services/ReportBuilder.java

```diff
@@ -0,0 +1,148 @@
+package com.csoft.services;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import org.apache.maven.project.MavenProject;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ReportBuilder {
+
+    private static final String PLUGIN_NAMESPACE = "com.github.carlomorelli:licensescan-maven-plugin";
```

Comment:
we can try to unit test this class. The two functions can be mocked using ByteArrayOutputStreams and Writer objects.